index.go

```diff
@@ -148,58 +148,18 @@ func (u *UUIDFieldIndex) PrefixFromArgs(args ...interface{}) ([]byte, error) {
 	}
 }
 
-func (u *UUIDFieldIndex) parseString(s string) ([]byte, error) {
+// parseString parses a UUID from the string. If enforceLength is false, it will
+// parse a partial UUID. An error is returned if the input, stripped of hyphens,
+// is not even length.
+func (u *UUIDFieldIndex) parseString(s string, enforceLength bool) ([]byte, error) {
 	// Verify the length
-	if len(s) != 36 {
+	if enforceLength && len(s) != 36 {
```

Comment:
This technically lets them supply a prefix that's larger, which is kind of weird, but I suppose it will never match anything.

I'd probably rename this `isFullUUID` and if that's true make sure length is exactly 36, otherwise make sure it's <= 36.